core/src/test/java/org/elasticsearch/ingest/core/CompoundProcessorTests.java

```diff
@@ -117,4 +111,68 @@ public void testSingleProcessorWithNestedFailures() throws Exception {
         assertThat(processorToFail.getInvokedCounter(), equalTo(1));
         assertThat(lastProcessor.getInvokedCounter(), equalTo(1));
     }
+
+    public void testCompoundProcessorExceptionFailWithoutOnFailure() throws Exception {
```

Comment:
haha, I don't blame you. I had a really awkward time trying to name all of these subtly different scenarios